projects/distributed-tracing/src/shared/components/span-detail/span-detail.component.ts

```diff
@@ -78,8 +78,8 @@ export class SpanDetailComponent implements OnChanges {
 
   public ngOnChanges(changes: TypedSimpleChanges<this>): void {
     if (changes.spanData) {
-      this.showRequestTab = !isEmpty(this.spanData?.requestHeaders) || !isEmpty(this.spanData?.requestBody);
-      this.showResponseTab = !isEmpty(this.spanData?.responseHeaders) || !isEmpty(this.spanData?.responseBody);
+      this.showRequestTab = !isEmpty(this.spanData?.requestHeaders) || !isEmpty(this.spanData?.requestBody$);
```

Comment:
this isEmpty check isn't meaningful on an observable. If we want to conditionally show the tab, then we need to fetch the data (could make this async), but that seems like it defeats the purpose of making it lazy.

Reply:
This field is not available in hypertrace. Hence this logic.

Reply:
This check seems to be what is causing a bug in waterfall. That check on the observable is always `true` since its an object, so the tabs are always showing.

Reply:
Oh yeah. Good review, Aaron!

Reply:
Should we convert the showRequestTab into an observable so that it can consume RequestBody$ correctly?

Reply:
Any reason why this won't work?

<img width="787" alt="Screen Shot 2021-08-06 at 1 57 50 PM" src="https://user-images.githubusercontent.com/45181984/128570308-703e2e56-41eb-4d6d-8ae3-4372ec1019f0.png">

Reply:
You may have to trigger change detection, idk. 

But this got me thinking. Would we ever have a case when request header is not present but requestBody will be? I would think not. So, imo we can just check presence of requestHeaders. and we wont need observables or subscription

Reply:
This seems to work in my testing so far. Going to run with it.

Reply:
@jake-bassett please check my latest comment

Reply:
Ah, just saw your last couple comments.

I agree with your assumptions about not having a body without headers. That is probably safe... buuuuut... I have tested the above code some more and am fairly confident this works for the case of a body with no headers. Not that we will likely ever run into it, but I see no harm in accounting for it if we already have the code.

Reply:
It's a double whammy - it both brings in a code subscribe rather than in the template (nooo!) but more relevant here, it makes the query eager, which I thought is exactly what this whole change is to avoid, I thought? We shouldn't be fetching until the tab is clicked on - the hiding logic being no longer relevant. If we think that there are scenarios where we won't even query this data (e.g. the attribute isn't present), then we need to make the whole property optional so we don't have to subscribe to it to find that out).